src/components/Dialog/Dialog.tsx

```diff
@@ -351,13 +356,29 @@ export default class Dialog extends PureComponent<DialogProps, DialogState> {
   }
 
   isShowTrigger(eventName: HideShowEvent) {
-    const { showTrigger } = this.props;
-    return convertToArray(showTrigger).indexOf(eventName) > -1;
+    const { showTrigger, addKeyboardHideShowTriggersByDefault } = this.props;
```

Comment:
this one feel a bit hard coded but I guess there is no prettier solution to do  now.
Maybe when we will add enter event support we could refactor it a bit.